media-gfx/tuxpaint: add 0.9.29, switch to sdl2, add new package media-libs/sdl2-pango, bump media-libs/tuxpaint-stamps

rndxelement opened the pull request:
Tux Paint 0.9.29 contains lots of [changes](https://tuxpaint.org/docs/CHANGES.php), including only supporting sdl2 from now on.
I therefore wrote the media-gfx/tuxpaint-0.9.29 ebuild such that it used sdl2.
Notes:
- Tux Paint needs a pango connection to sdl2. Previously this was media-libs/sdl-pango. Switching to sdl2 I noticed that there is no ebuild for media-libs/sdl2-pango. That's why I added it as a new package.
- In the new media-libs/sdl2-pango ebuild I added the `ppc` keyword, even though I'm not able to test it. Without this keyword it would not be possible to push the media-gfx/tuxpaint-0.9.29 ebuild as it supports `ppc`. I hope this is okay.
- I version bumped media-gfx/tuxpaint-stamps.
- Tested all three packages in a minimal amd64 stage-3 chroot with emerge and pkg-testing-tool
- Runtime tested Tux Paint; seems to work fine; I saw new features and the new stickers integrate well

rndxelement commented:
Rebased to latest master commit.

juippis commented on media-libs/sdl2-pango/sdl2-pango-2.1.5.ebuild:
```diff
@@ -0,0 +1,33 @@
+LICENSE="LGPL-2.1"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~x86"
+IUSE="static-libs"
```
Is this needed? Nothing seems to depend on this library with static-lib requirement, so it should just be disabled.
https://projects.gentoo.org/qa/policy-guide/installed-files.html#pg0302

Also:
https://projects.gentoo.org/qa/policy-guide/installed-files.html#pg0303
but I see you're handling `src_install` properly already.

juippis commented on media-libs/sdl2-pango/sdl2-pango-2.1.5.ebuild:
```diff
@@ -0,0 +1,33 @@
+	media-libs/libsdl2[video]
+	media-libs/freetype"
+DEPEND="${RDEPEND}
+	virtual/pkgconfig"
```
This is a BDEPEND instead.

juippis commented on media-libs/sdl2-pango/sdl2-pango-2.1.5.ebuild:
```diff
@@ -0,0 +1,33 @@
+HOMEPAGE="https://github.com/markuskimius/SDL2_Pango/"
+SRC_URI="https://github.com/markuskimius/SDL2_Pango/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="LGPL-2.1"
```
Oh and this is `LGPL-2.1+`, 

```
/*  SDL_Pango.c -- A companion library to SDL for working with Pango.
    Copyright (C) 2004 NAKAMURA Ken'ichi

    This library is free software; you can redistribute it and/or
    modify it under the terms of the GNU Lesser General Public
    License as published by the Free Software Foundation; either
    version 2.1 of the License, or (at your option) any later version.
...
```

rndxelement commented:
@juippis Thanks for the review and all the helpful suggestions!
Changes:
- Dropped all keywords except ~amd64
- Adjusted the license for media-libs/sdl2-pango
- Removed static-libs from media-libs/sdl2-pango
- Reordered the dependencies and moved pkgconfig to BDEPEND in media-libs/sdl2-pango  

I build and runtime tested TuxPaint and TuxPaint-stamps, as well as sdl2-pango. Also rebased to current master.

juippis commented on media-libs/sdl2-pango/sdl2-pango-2.1.5.ebuild:
```diff
@@ -0,0 +1,25 @@
+
+src_install() {
+	default
+	find "${ED}" -name "*.a" -delete || die "Failed to delete .a files"
```
Expand this to delete **.la** files too,
```
 * QA Notice: Installing libtool files (.la) without corresponding static libraries!
 *   /usr/lib64/libSDL2_Pango.la
```

juippis commented on media-libs/sdl2-pango/sdl2-pango-2.1.5.ebuild:
```diff
@@ -0,0 +1,25 @@
+SLOT="0"
+KEYWORDS="~amd64"
+
+RDEPEND="media-libs/freetype
+	media-libs/libsdl2[video]
+	x11-libs/pango"
```
Looks like you need more deps,
```
 * VDB: detected possibly incorrect RDEPEND (media-libs/sdl2-pango-2.1.5)
 *  > dev-libs/glib:2
 *  > media-libs/fontconfig
 *  > media-libs/harfbuzz:=
```

juippis reviewed (commented):
I couldn't install tuxpaint due to:
```
make -j32 CC=x86_64-pc-linux-gnu-gcc GENTOO_LIBDIR=lib64
echo


echo "...Compiling i18n support..."



...Generating 'Magic' tool API development header file...
...Compiling i18n support...
x86_64-pc-linux-gnu-gcc  -O0 -g -W -Wall -fno-common -ffloat-store -fvisibility=hidden -Wcast-align -        Wredundant-decls -Wbad-function-cast -Wwrite-strings -Waggregate-return -Wstrict-prototypes -Wmissing-       prototypes -Wstrict-aliasing=2 -I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/       include -I/usr/lib64/libffi/include -I/usr/include/harfbuzz -I/usr/include/freetype2 -pthread -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -I/usr/include/cairo -I/usr/include/libpng16 -I/usr/    include/pixman-1  -march=native -O2 -pipe -frecord-gcc-switches  -DVER_DATE=\"2023-05-28\" -DVER_VERSION=\"0.9.29\" -DDATA_PREFIX=\"/usr/share/tuxpaint/\" -DDOC_PREFIX=\"/usr/share/doc/tuxpaint-0.9.29/\" -             DLOCALEDIR=\"/usr/share/locale/\" -DIMDIR=\"/usr/share/tuxpaint/im/\" -DCONFDIR=\"/etc/tuxpaint/\" -         DMAGIC_PREFIX=\"/usr/lib64/tuxpaint/plugins/\"       \
    -c src/i18n.c -o obj/i18n.o

...Compiling IM support...
...Compiling pixel functions...

...Compiling RGB to Linear functions...


...Compiling sound playback functions...

...Compiling cursor support...
...Compiling font support...

...Generating the command-line and config file parser (STEP 1)...
...Compiling flood fill tool...
...Compiling progress bar functions...

...Compiling directory-walking support...



...Compiling filename support...

msgfmt -o trans/ach.mo src/po/ach.po
...Compiling animated GIF export libary...

...Compiling on screen keyboard support...
...Compiling sound effect list...
msgfmt -o trans/af.mo src/po/af.po
...Compiling PostScript print support...
msgfmt -o trans/ak.mo src/po/ak.po
msgfmt -o trans/am.mo src/po/am.po
msgfmt -o trans/an.mo src/po/an.po
msgfmt -o trans/ar.mo src/po/ar.po
...Preparing translation files...
msgfmt: error while opening "trans/ach.mo" for writing: No such file or directory
make: *** [Makefile:538: trans/ach.mo] Error 1
make: *** Waiting for unfinished jobs....
In file included from /usr/include/stdio.h:906,
                 from src/get_fname.c:27:
In function 'snprintf',
    inlined from 'get_fname' at src/get_fname.c:107:3:
/usr/include/bits/stdio2.h:54:10: warning: 'dir' may be used uninitialized [-Wmaybe-uninitialized[https://   gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wmaybe-uninitialized]]
   54 |   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   55 |                                    __glibc_objsize (__s), __fmt,
      |                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   56 |                                    __va_arg_pack ());
      |                                    ~~~~~~~~~~~~~~~~~
src/get_fname.c: In function 'get_fname':
src/get_fname.c:92:15: note: 'dir' was declared here
   92 |   const char *dir;
      |               ^~~
src/fonts.c: In function 'TuxPaint_Font_OpenFont':
src/fonts.c:369:16: warning: assignment discards 'const' qualifier from pointer target type [-Wdiscarded-    qualifiers[https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wdiscarded-qualifiers]]
  369 |     familyname = TTF_FontFaceFamilyName(ttf_font);      /* N.B.: I don't believe we're supposed to   free() this... -bjk 2021.10.26 */
      |                ^
src/fonts.c: In function 'getfonthandle':
src/fonts.c:1419:20: warning: the comparison will always evaluate as 'true' for the address of 'filename'    will never be NULL [-Waddress[https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Waddress]]
 1419 |   if (fi->filename != NULL)
      |                    ^~
In file included from src/fonts.c:60:
src/fonts.h:175:9: note: 'filename' declared here
  175 |   char *filename[4];
      |         ^~~~~~~~
 * ERROR: media-gfx/tuxpaint-0.9.29::gentoo failed (compile phase):
 *   emake failed
```

juippis reviewed (approved):
```
 * QA Notice: Files built without respecting CFLAGS have been detected
 *  Please include the following list of files in your report:
 * /usr/share/doc/tuxpaint-0.9.29/en/tp_magic_example.so
 * QA Notice: Files built without respecting LDFLAGS have been detected
 *  Please include the following list of files in your report:
 * /usr/share/doc/tuxpaint-0.9.29/en/tp_magic_example.so
```
Thanks!

juippis commented:
Please request re-keywording through bugzilla if you feel these arches should be supported. Let me know if you're not aware how to do that exactly.

rndxelement commented:
> 

Okay, thanks a lot!
